src/openms/source/ANALYSIS/OPENSWATH/IonMobilityScoring.cpp

```diff
@@ -200,6 +210,141 @@ namespace OpenMS
     }
   }
 
+  OpenMS::Mobilogram sumAlignedMobilograms(const std::vector<OpenMS::Mobilogram>& aligned_mobilograms) 
+  {
+    if (aligned_mobilograms.empty()) {
+      return OpenMS::Mobilogram();
+    }
+
+    OpenMS::Mobilogram summed_mobilogram;
+
+    // Use the first mobilogram to set the structure
+    const auto& first_mobilogram = aligned_mobilograms[0];
+
+    for (size_t j = 0; j < first_mobilogram.size(); ++j) {
+      OpenMS::MobilityPeak1D summed_peak;
+      summed_peak.setMobility(first_mobilogram[j].getMobility());
+      summed_peak.setIntensity(0.0); 
+
+      // Sum intensities from all mobilograms
+      for (const auto& mobilogram : aligned_mobilograms) {
+        if (j < mobilogram.size()) {
+          summed_peak.setIntensity(summed_peak.getIntensity() + mobilogram[j].getIntensity());
+        }
+      }
+
+      summed_mobilogram.push_back(summed_peak);
+    }
+
+    // Concatenate names of all mobilograms
+    std::string concatenated_name;
+    for (size_t i = 0; i < aligned_mobilograms.size(); ++i) {
+      if (i > 0) {
+        concatenated_name += "|";
+      }
+      concatenated_name += aligned_mobilograms[i].getName();
+    }
+
+    // Set the concatenated name for the summed mobilogram
+    summed_mobilogram.setName(concatenated_name);
+
+    return summed_mobilogram;
+  }
+
+  std::tuple<size_t, size_t, size_t> findHighestPeak(const PeakPickerMobilogram& picker) 
+  {
+    const auto& intensities = picker.integrated_intensities_;
+
+    // Check if vectors are empty or of different sizes
+    if (intensities.empty() ||
+        intensities.size() != picker.left_width_.size() ||
+        intensities.size() != picker.right_width_.size()) {
+      // Return an "invalid" tuple if there's an issue
+      return std::make_tuple(std::numeric_limits<size_t>::max(), 0, 0);
+    }
+
+    // Find the iterator pointing to the maximum element
+    auto max_it = std::max_element(intensities.begin(), intensities.end());
+
+    // Get the index of the maximum element
+    size_t max_index = std::distance(intensities.begin(), max_it);
+
+    // Return the tuple
+    return std::make_tuple(max_index,
+                            picker.left_width_[max_index],
+                            picker.right_width_[max_index]);
+  }
+
+  void filterPeakIntensities(OpenMS::Mobilogram& mobilogram,
+                               size_t left_index,
+                               size_t right_index) 
+  {
+    // Check if indices are valid (indicating peaks were found)
+    bool peaksFound = (left_index != std::numeric_limits<size_t>::max() &&
```

Comment:
I removed the peaksFound check in `filterPeakIntensities`. I initially had this check because sometimes the XIM only has a single point, which the peak picking would return 0 for both the leftwidth and rightwidth. I have added another condition in the if statement to check if the XIM is not empty and if it's not a single point. 

I'm not sure if I should explicitly say it shouldn't be 1, or should be at least a size of N? Because the PeakPicker probably assumes that it's going to pick on data that's not just a single point? 

```
if ( apply_im_peak_picking ) 
{
    PeakPickerMobilogram::PeakPositions peak_pos{};
    // Ion mobilogram cannot be empty and cannot have a single point
    if ( !aligned_ms2_mobilograms.empty() && aligned_ms2_mobilograms[0].size()!=1 )
    {
      // Perform peak picking ...
    }
    else
    {
    // Skip peak picking for this XIM
     }
}   
```